### PR TITLE
Add savepoint failure tests

### PR DESCRIPTION
**Pull request by jasonmp85**
Tests at each significant point (i.e. SAVEPOINT, ROLLBACK, RELEASE) that correct semantics are preserved (using both no and statement replication).

### CODE REVIEW

**Comment by jasonmp85** [comment]
OK, so this clearly has a _lot_ of output, but in sequences of savepoints, rollbacks, releases, etc., I wanted to test the significant points during the transaction. Additionally, I needed to test under no replication and statement replication.

This _doesn't_ test behavior of error handling under subtransactions (something the main tests devote a lot of space to), but it still ended up being pretty large. Covers existing failure tests, as of the 7.4 release documents (the most recent one which mentioned `SAVEPOINT` in detail).

File: src/test/regress/sql/failure_savepoints.sql [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+);
+SELECT create_distributed_table('artists', 'id');
+
+-- add some data
+INSERT INTO artists VALUES (1, 'Pablo Picasso');
+INSERT INTO artists VALUES (2, 'Vincent van Gogh');
+INSERT INTO artists VALUES (3, 'Claude Monet');
+INSERT INTO artists VALUES (4, 'William Kurelek');
+
+SELECT citus.clear_network_traffic();
```
Comment (thanodnl) [inline src/test/regress/sql/failure_savepoints.sql]: nit; I don't think this is required for the tests if you don't list the recorded network traffic

File: src/test/regress/sql/failure_savepoints.sql [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+ROLLBACK TO s2;
+RELEASE SAVEPOINT s2;
+COMMIT;
+
+SELECT * FROM artists WHERE id=7;
+
+SELECT citus.mitmproxy('conn.onQuery(query="^ROLLBACK").kill()');
+
+-- Recover from errors
+\set VERBOSITY terse
```
Comment (thanodnl) [inline src/test/regress/sql/failure_savepoints.sql]: this statement is not undone, is it expected to run the rest of the suite in this mode?

**Comment by sumedhpathak** [comment]
@thanodnl Same note as the other one. Could you get this change merged in after applying your feedback if any? Thanks!

**Comment by thanodnl** [comment]
Same as other one, will squash/rebase/land once tests run again

**Pull request merged**